### PR TITLE
Document setting (sub) title in the app guide

### LINKED ISSUE #1577: Document setting title and sub title
We need to cover how to set the title and sub title on an application. This includes the `TITLE` and `SUB_TITLE` class vars, but also the instance attributes `title` and `sub_title`.

This probably belongs in the "App Basics" page in the guide, near the top.

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
Fix #1577

### CODE REVIEW

**Review by davep — APPROVED** [review]

File: docs/guide/app.md [hunk 1]
```diff
@@ -147,7 +148,39 @@ The addition of `[str]` tells mypy that `run()` is expected to return a string.
 
     Type annotations are entirely optional (but recommended) with Textual.
 
-## CSS
+## Customising your app
```
Comment (willmcgugan) [inline docs/guide/app.md:151]: Not sure this hierarchy is required. Can we put the following section under "App Class" after the run method.

File: docs/guide/app.md [hunk 2]
```diff
@@ -147,7 +148,39 @@ The addition of `[str]` tells mypy that `run()` is expected to return a string.
...
+
+### Setting the title and subtitle
+
+By default, the title of your application matches the name of the application class and the subtitle is empty.
```
Comment (willmcgugan) [inline docs/guide/app.md:155]: It's perhaps not clear to the reader yet what "title" and "subtitle" are. Suggest an introduction something along these line:

"Textual apps have a "title" attribute which is the name of your app, and a "subtitle" attribute which is additional title text that indicates the context of your app. Both values are will be displayed in the apps Header (if there is one)."

**Review by willmcgugan — CHANGES_REQUESTED** [review]
Examples are great. The text could use a little revision.

File: docs/guide/app.md (hunk 1 above)
Comment (willmcgugan) [reply docs/guide/app.md:151]: Come to think of it. If you are mentioning compose, it likely should come later.

I think the "Title and Subtitle" should come after CSS. On a top-level heading of its own.

Just noticed another unrelated issue. "Exiting" is under Widgets. But that feels wrong. Could you promote Exiting to a top header.

Comment (rodrigogiraoserrao) [reply docs/guide/app.md]: Addressed in 476eca7.

File: docs/guide/app.md [hunk 3]
```diff
@@ -182,6 +185,41 @@ Here's the question app with classvar CSS:
...
+
+For example, the application shown below changes its title and subtitle as soon as the application is instantiated.
+
+```py title="question_title02.py" hl_lines="19-21"
+--8<-- "docs/examples/app/question_title02.py"
+```
+
+If you run the app shown above and if you press a key, the title and subtitle update accordingly.
+
+For example, if you press `t`, your application will look as shown below:
```
Comment (willmcgugan) [inline docs/guide/app.md:217]: We have an extension that renders keys, try replacing the key with `++t++`

**Review by willmcgugan — APPROVED** [review]

File: docs/guide/app.md (hunk 3 above)
Comment (rodrigogiraoserrao) [reply docs/guide/app.md]: Thanks for the reminder o7

**Pull request merged**